### PR TITLE
Support updating AAAA records

### PR DESCRIPTION
**Pull request by Mygod**
Some possible enhancements that I do not currently plan to do:

* ~Support Windows~ (I don't think this repo has ever considered supporting Windows but this thing definitely does not)
* Multiple (either temporary/dynamic or renewed) IPv6 addresses

### CODE REVIEW

File: cloudflare-ddns [hunk 1]
```diff
@@ -89,13 +91,49 @@ def get_external_ip():
     return requests.get(EXTERNAL_IP_QUERY_API).json()['ip']
 
 
-def update_dns(subdomain, auth, ip_address):
+def get_ipv6():
+    """
+    Based on: https://gist.github.com/corny/7a07f5ac901844bd20c9
+
+    :return: A string representing the network's
```
Comment (LINKIWI) [inline cloudflare-ddns]: Incomplete thought?

File: cloudflare-ddns [hunk 2]
```diff
@@ -1,13 +1,15 @@
 #!/usr/bin/python
 import os
```
Comment (LINKIWI) [inline cloudflare-ddns:4]: Nit: please alphabetize all imports

File: cloudflare-ddns [hunk 3]
```diff
@@ -111,30 +149,20 @@ def update_dns(subdomain, auth, ip_address):
...
-        print('We couldn\'t find an A record for the (sub)domain {subdomain}. Are you sure you\'ve created one?'.format(subdomain=subdomain))
-        return
-    dns_record_id = dns_records[0]['id']
-    existing_ip_address = dns_records[0]['content']
+    # Find DNS records
+    record_a = None
+    record_aaaa = None
+    for dns_record in requests.get(CLOUDFLARE_ZONE_DNS_RECORDS_QUERY_API.format(zone_id=zone_id), headers=auth, params={'name': subdomain}).json()['result']:
+        if dns_record['type'] == 'A':
+            assert(record_a is None)
```
Comment (LINKIWI) [inline cloudflare-ddns]: Consider logging/printing an error message here to reduce confusion for the user

File: cloudflare-ddns [hunk 4]
```diff
@@ -111,30 +149,20 @@ def update_dns(subdomain, auth, ip_address):
...
-            CLOUDFLARE_ZONE_DNS_RECORDS_UPDATE_API.format(zone_id=zone_id, dns_record_id=dns_record_id),
-            headers=dict(list(auth.items()) + [('Content-Type', 'application/json')]),
-            data=json.dumps(new_dns_record),
-        )
-        if update_resp.json()['success']:
-            print('DNS record updated successfully!')
-        else:
-            print('DNS record failed to update.\nCloudFlare returned the following errors: {errors}.\nCloudFlare returned the following messages: {messages}'.format(errors=update_resp.json()['errors'], messages=update_resp.json()['messages']))
+    update_dns_record(auth, zone_id, record_a, ipv4_address)
+    update_dns_record(auth, zone_id, record_aaaa, ipv6_address)
```
Comment (LINKIWI) [inline cloudflare-ddns:162]: Can you guard IPv6/AAAA record functionality behind a flag?

**Review by LINKIWI — COMMENTED** [review]
I don't use Windows so it's totally fine that this patch does not support Windows.

File: cloudflare-ddns (hunk 3 above)
Comment (Mygod) [reply cloudflare-ddns]: I don't think this assertion will fail ever.

EDIT: As long as CloudFlare doesn't change their API that is.

File: cloudflare-ddns (hunk 4 above)
Comment (Mygod) [reply cloudflare-ddns:162]: If there isn't an AAAA record existing, it wouldn't do anything.

**Comment by Mygod** [comment]
I tested it and so far it seems to work perfectly with SLAAC with privacy extensions. It should also work with DHCPv6 where client addresses are like `inet6 abcd::dcba/128 scope global`.

Everything should be good now? I guess I could also test it some more for a few days. @LINKIWI

**Comment by LINKIWI** [comment]
lgtm

**Comment by Mygod** [comment]
Well that assertion is there just in case Cloudflare changed stuff but okay I guess. :)

**Pull request merged**